qa/rpc-tests/test_framework/bumessages.py

```diff
@@ -0,0 +1,405 @@
+        self.addrFromPort = struct.unpack("<H", f.read(2))[0]
+        return self
+
+    def serialize(self):
+        r = b""
+        r += struct.pack("<H", self.addrFromPort)
+        return r
+
+    def __repr__(self):
+        return "msg_buversion(addrFromPort=%d)" % (self.addrFromPort)
+
+
+class msg_buverack(object):
+    command = b"buverack"
+
+    def __init__(self):
+        pass
+
+    def deserialize(self, f):
+        return self
+
+    def serialize(self):
+        r = b""
+        return r
+
+    def __repr__(self):
+        return "msg_buverack()"
+
+
+class QHash(object):
+    """quarter hash"""
+
+    def __init__(self, shortHash=None):
+        self.hash = shortHash
+
+    def deserialize(self, f):
+        self.hash = struct.unpack("<Q", f.read(8))[0]
+        return self
+
+    def serialize(self):
+        r = b""
+        r += struct.pack("<Q", self.hash)
+        return r
+
+    def __repr__(self):
+        return "QHash(0x%016x)" % (self.hash)
+
+
+class Hash(object):
+    """sha256 hash"""
+
+    def __init__(self, hash=None):
+        self.hash = hash
+
+    def deserialize(self, f):
+        self.hash = deser_uint256(f)
+        return self
+
+    def serialize(self):
+        r = b""
+        r += ser_uint256(self.hash)
+        return r
+
+    def __str__(self):
+        return "%064x" % self.hash
+
+    def __repr__(self):
+        return "Hash(%064x)" % self.hash
+
+
+class CXThinBlock(CBlockHeader):
+    def __init__(self, header=None, vTxHashes=None, vMissingTx=None):
+        super(CXThinBlock, self).__init__(header)
+        self.vTxHashes = vTxHashes
+        self.vMissingTx = vMissingTx
+
+    def deserialize(self, f):
+        super(CXThinBlock, self).deserialize(f)
+        self.vTxHashes = deser_vector(f, QHash)
+        self.vMissingTx = deser_vector(f, CTransaction)
+        return self
+
+    def serialize(self):
+        r = b""
+        r += super(CXThinBlock, self).serialize()
+        r += ser_vector(self.vTxHashes)
+        r += ser_vector(self.vMissingTx)
+        return r
+
+    def summary(self):
+        s = []
+        s.append(super(self.__class__, self).summary())
+        s.append("\nQuarter Hashes")
+        count = 0
+        for qh in self.vTxHashes:
+            if (count % 5) == 0:
+                s.append("\n%4d: " % count)
+            s.append("%016x " % qh.hash)
+            count += 1
+
+        s.append("\nFull Transactions\n")
+        count = 0
+        for tx in self.vMissingTx:
+            s.append("%4d: %s\n" % (count, tx.summary()))
+            count += 1
+        return "".join(s)
+
+    def __str__(self):
+        return "CXThinBlock(nVersion=%i hashPrevBlock=%064x hashMerkleRoot=%064x nTime=%s nBits=%08x nNonce=%08x vTxHashes_len=%d vMissingTx_len=%d)" \
+            % (self.nVersion, self.hashPrevBlock, self.hashMerkleRoot, time.ctime(self.nTime), self.nBits, self.nNonce, len(self.vTxHashes), len(self.vMissingTx))
+
+    # too large to print
```

Comment:
I am trying to preserve symmetry with the other object classes defined in nodemessages.py.  However, in my setup, calling repr on a block results in such a painfully large single line printout that I was often forced to kill the python shell from another window.  But the problem is that its easy to inadvertently call repr.  By removing this function call, this gotcha is prevented, and I think that there is no current use for it.  However if such a use appears, it is valuable to have the function so it can be quickly re-enabled.